### PR TITLE
Add GELU fusion

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
Add 3 patterns that can be fused to GELU op.

### Tickets:
 - 50817

### CODE REVIEW

**Review by mateusztabaka — COMMENTED** [review]
I think you should fix indentation - in some places you're probably using two spaces for tab instead of four.

File: inference-engine/src/transformations/include/transformations/common_optimizations/gelu_fusion.hpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+class TRANSFORMATIONS_API GeluFusionWithErfTwo;
+class TRANSFORMATIONS_API GeluFusionWithErfThree;
+
+} // namespace pass
+} // namespace ngraph
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief GeluFusion transformation replaces a sub-graph (0.5 * x) * (1 + erf(x
+ * / sqrt(2))) with a Gelu op.
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply inference-engine/src/transformations/include/transformations/common_optimizations/gelu_fusion.hpp:27]: Ok, my bad - that was automatic next-line-move by formatter which I looked over

File: inference-engine/tests/functional/inference_engine/transformations/gelu_fusion.cpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    {
+        auto data = std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto gelu = std::make_shared<opset7::Gelu>(data);
+        f_ref = std::make_shared<Function>(NodeVector{gelu}, ParameterVector{data});
+    }
+
+    auto res = compare_functions(f, f_ref);
+    ASSERT_TRUE(res.first) << res.second;
+}
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply inference-engine/tests/functional/inference_engine/transformations/gelu_fusion.cpp:144]: Done

File: inference-engine/tests/functional/inference_engine/transformations/gelu_fusion.cpp [hunk 3]
```diff
@@ -142,3 +142,83 @@ TEST(TransformationTests, GeluFusionPatternThree) {
...
+    {
+        auto data =
+            std::make_shared<opset1::Parameter>(element::f32, Shape{2, 2});
+        auto gelu = std::make_shared<opset7::Gelu>(data);
+        f_ref =
+            std::make_shared<Function>(NodeVector{gelu}, ParameterVector{data});
+    }
+
+    auto res = compare_functions(f, f_ref);
+    ASSERT_FALSE(res.first) << res.second;
```
Comment (mateusztabaka) [inline inference-engine/tests/functional/inference_engine/transformations/gelu_fusion.cpp:223]: I think it's better to have reference function to be a copy of the original and put ASSERT_TRUE here.

**Review by GlebKazantaev — APPROVED** [review]

**Pull request merged**